library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileOutputOperator.java

```diff
@@ -484,34 +484,40 @@ public FSFilterStreamContext load(@Nonnull String filename)
         //of this operator.
         for(String seenFileName: endOffsets.keySet()) {
           try {
-            Integer part = openPart.get(seenFileName).getValue() + 1;
+            Integer fileOpenPart = this.openPart.get(seenFileName).getValue();
+            int nextPart = fileOpenPart + 1;
```

Comment:
Regd 1. good point. However, if there was a fault recovery the tmp filename for a part will change and later when it becomes eligible for finalization as you mentioned in committed callback the new tmp filename will be moved, the old tmp file would still remain isn't it.

Regd 2. if there is a failure the lease would be held by a prior instance of this operator and the new instance would try to write to the same file as the tmp file isn't being recreated.